xml-parser/src/test/java/org/uniprot/core/xml/unirule/CommonConverterTest.java

```diff
@@ -0,0 +1,183 @@
+package org.uniprot.core.xml.unirule;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.jvnet.jaxb2_commons.lang.Equals2;
+import org.uniprot.core.RangeTest;
+import org.uniprot.core.uniprotkb.description.impl.ProteinDescriptionBuilderTest;
+import org.uniprot.core.uniprotkb.impl.UniProtKBAccessionBuilderTest;
+import org.uniprot.core.unirule.impl.*;
+import org.uniprot.core.xml.Converter;
+import org.uniprot.core.xml.jaxb.unirule.AnnotationType;
+
+public class CommonConverterTest {
```

Comment:
I can see what you're doing. The difficulty with reflection is that it's hard to read (e.g., methods not obvious) and therefore I can't really comment here ... I would suggest asking someone else to look at this too.